src/Shared/CommunicationsUtilities.cs

```diff
@@ -388,7 +384,7 @@ internal static void WriteLongForHandshake(this PipeStream stream, long value)
         /// Extension method to read a series of bytes from a stream.
         /// If specified, leading byte matches one in the supplied array if any, returns rejection byte and throws IOException.
         /// </summary>
-        internal static long ReadLongForHandshake(this PipeStream stream, byte[] leadingBytesToReject,
+        internal static long ReadLongForHandshake(this PipeStream stream,
```

Comment:
Since this handshake has caused some pain, maybe consider in the exceptions (like the one below) including the hex form of the bytes that were received (and possibly what was expected, if it was different)